core/capabilities/ccip/ccipsolana/commitcodec.go

```diff
@@ -0,0 +1,138 @@
+package ccipsolana
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"math/big"
+
+	agbinary "github.com/gagliardetto/binary"
+	"github.com/gagliardetto/solana-go"
+
+	"github.com/smartcontractkit/chainlink-ccip/chains/solana/gobindings/ccip_router"
+	"github.com/smartcontractkit/chainlink-ccip/chains/solana/utils/common"
+	cciptypes "github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
+)
+
+// CommitPluginCodecV1 is a codec for encoding and decoding commit plugin reports.
+// Compatible with:
+// - "OffRamp 1.6.0-dev"
+type CommitPluginCodecV1 struct{}
+
+func NewCommitPluginCodecV1() *CommitPluginCodecV1 {
+	return &CommitPluginCodecV1{}
+}
+
+func (c *CommitPluginCodecV1) Encode(ctx context.Context, report cciptypes.CommitPluginReport) ([]byte, error) {
+	var buf bytes.Buffer
+	encoder := agbinary.NewBorshEncoder(&buf)
+	mr := ccip_router.MerkleRoot{}
+
+	if len(report.MerkleRoots) != 0 {
```

Comment:
curious if we should do validation that there's only 1 MerkleRoot in here?

Reply:
Yes I had for execute, adding it for commit as well